test/utils/Utils.test.ts

```diff
@@ -3,18 +3,119 @@
 
 import * as chai from 'chai';
 
-import { wait } from '../../src/utils/Utils';
+import { toLowerCasePropertyNames, wait } from '../../src/utils/Utils';
 
 describe('Utils', () => {
   const expect: Chai.ExpectStatic = chai.expect;
 
-  it('atleast waits for the specified delay in milliseconds', async () => {
-    const startTime = new Date().getTime();
-    const delay = 500;
-    await wait(delay);
-    const endTime = new Date().getTime();
+  describe('wait', () => {
+    it('atleast waits for the specified delay in milliseconds', async () => {
+      const startTime = new Date().getTime();
+      const delay = 500;
+      await wait(delay);
+      const endTime = new Date().getTime();
 
-    // Timing is imprecise. Allow 2 milliseconds of fudge. I have seen this run in 499.
-    expect(endTime - startTime).to.gte(delay - 2);
+      // Timing is imprecise. Allow 2 milliseconds of fudge. I have seen this run in 499.
+      expect(endTime - startTime).to.gte(delay - 2);
+    });
+  });
+
+  describe('toLowerCasePropertyNames', () => {
+    it('supports null as an input', () => {
+      const input: null = null;
+      const output = toLowerCasePropertyNames(input);
+      expect(output).to.eq(null);
+    });
+
+    it('supports non object data as an input', () => {
+      const input = 1;
+      const output = toLowerCasePropertyNames(input);
+      expect(output).to.eq(1);
+    });
+
+    it('supports an empty object as an input', () => {
+      const input = {};
+      const output = toLowerCasePropertyNames(input);
+      expect(output).to.be.empty;
+    });
+
+    it('supports an undefined object as an input', () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const input: any = undefined;
+      const output = toLowerCasePropertyNames(input);
+      expect(output).to.eq(undefined);
+    });
+
+    it('converts upper case keys to lower case', () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const input: any = {
+        MeetingId: 'meeting-id',
+        MediaPlacement: {
+          AudioHostUrl: 'audio-host-url',
+          ScreenDataUrl: 'screen-data-url',
+          ScreenSharingUrl: 'screen-sharing-url',
+          ScreenViewingUrl: 'screen-viewing-url',
+          SignalingUrl: 'signaling-url',
+          TurnControlUrl: 'turn-control-url',
+        },
+        Http: [[]],
+      };
+
```

Comment:
I could be missing something, but shouldn't the input consist of upper case letters as we're trying to test converting upper case to lower case?

Reply:
The input on line 61 is testing for the scenario when we have two empty arrays. It is important I add that to the test as there is a if block in the method that was the root cause and I am testing against it. Also, the keys will be converted to lowercase and not the values. So, theoretically `Http` will become `http`.

Reply:
Ah, I thought it was changing the values to lower case, but yeah that makes sense.